mlir/lib/Dialect/Affine/Analysis/AffineStructures.cpp

```diff
@@ -1803,19 +1812,40 @@ LogicalResult mlir::getRelationFromMap(AffineMap &map,
       eq[j + numRangeVars] = flatExprs[i][j];
     // Set this dimension to -1 to equate lhs and rhs and add equality.
     eq[numDomainVars + i] = -1;
+    if (!accessRange.empty() && accessRange[i] != 0)
+      eq[numDimSymLocalVars + i] = 1;
     localVarCst.addEquality(eq);
   }
 
+  // Add inequalities of the local var
+  SmallVector<int64_t, 8> ineq(numCols, 0);
+  for (unsigned i = 0, j = 0, e = accessRange.size(); i < e; ++i) {
+    if (accessRange[i] == 0) {
+      break;
+    }
+    // Zero fill.
+    std::fill(ineq.begin(), ineq.end(), 0);
+    // Fill Inequality
+    ineq[numDimSymLocalVars + j] = 1;
+    ineq[numCols - 1] = 0;
+    localVarCst.addInequality(ineq);
+    ineq[numDimSymLocalVars + j] = -1;
+    ineq[numCols - 1] = accessRange[i] - 1;
+    localVarCst.addInequality(ineq);
+    j++;
```

Comment:
This could be moved to Line 1822 as part of the loop increment

Reply:
there is early exit, so j don't increase sometimes